File: BetterPlacemaking.CLIENT/src/app/models/FusionDtos.ts
export interface FusionRunDto {
  Id: string;
  Status: 'running' | 'cancelling' | 'cancelled' | 'success' | 'failed' | 'unknown';
  TriggeredBy: 'manual' | 'scheduled' | 'unknown';
  FromDateUnix?: number;
  ToDateUnix?: number;
  StartedAtUnix?: number;
  CompletedAtUnix?: number;
  RecordsFused?: number;
  ErrorMessage?: string;
  OutputGcsPath?: string;
  ProjectId?: string;
}

export interface FusionConfigDto {
  ScheduledHourUtc: number;
  ScheduledMinuteUtc: number;
  Enabled: boolean;
  ProjectId?: string;
}

export interface TriggerFusionDto {
  FromDateUnix: number;
  ToDateUnix: number;
  ProjectId?: string;
}

export interface UpdateFusionConfigDto {
  ScheduledHourUtc: number;
  ScheduledMinuteUtc: number;
  Enabled: boolean;
  ProjectId?: string;
}

File: BetterPlacemaking.SERVER/Models/Dtos/FusionDtos.cs
namespace BetterPlacemaking.Models.Dtos.Fusion
{
    public record FusionRunDto(
        string Id,
        string Status,
        string TriggeredBy,
        double? FromDateUnix,
        double? ToDateUnix,
        double? StartedAtUnix,
        double? CompletedAtUnix,
        int? RecordsFused,
        string? ErrorMessage,
        string? OutputGcsPath = null,
        string? ProjectId = null
    );

    public record FusionConfigDto(
        int ScheduledHourUtc,
        int ScheduledMinuteUtc,
        bool Enabled,
        string? ProjectId = null
    );

    public record TriggerFusionDto(
        double FromDateUnix,
        double ToDateUnix,
        string? ProjectId = null
    );

    public record UpdateFusionConfigDto(
        int ScheduledHourUtc,
        int ScheduledMinuteUtc,
        bool Enabled,
        string? ProjectId = null
    );
}
